control-plane/subcommand/common/common.go

```diff
@@ -115,3 +118,44 @@ func WriteFileWithPerms(outputFile, payload string, mode os.FileMode) error {
 	}
 	return os.Chmod(outputFile, mode)
 }
+
+// GetHTTPSScheme return the string value "https" if the bool https is true and "http" if it is false.
+func GetHTTPSScheme(https bool) string {
+	if https {
+		return "https"
+	} else {
+		return "http"
+	}
+}
+
+// GetResolvedServerAddresses resolves the Consul server address if it has been provided a provider else it returns the server addresses that were input to it.
+func GetResolvedServerAddresses(serverAddresses []string, providers map[string]discover.Provider, logger hclog.Logger) ([]string, error) {
+	var err error
+	if len(serverAddresses) == 1 && strings.Contains(serverAddresses[0], "provider=") {
+		serverAddresses, err = godiscover.ConsulServerAddresses(serverAddresses[0], providers, logger)
+		if err != nil {
+			logger.Error(fmt.Sprintf("Unable to discover any Consul addresses from %q: %s", serverAddresses[0], err))
+			return nil, err
+		}
+	}
+	return serverAddresses, nil
```

Comment:
```suggestion
	if len(serverAddresses) != 1 || !strings.Contains(serverAddresses[0], "provider=") {
		return serverAddresses, nil
	}
	return godiscover.ConsulServerAddresses(serverAddresses[0], providers, logger)
```

just nits, take em or leave em:
1. Don't see the need to re-use the serverAddresses variable, can redefine a new one if you end up using godiscover
2. logging makes sense to me in the caller. So basically functions return errs and it's up to the callers to log. Otherwise you end up with duplicate logs (both caller and callee) or you end up diving into functions so see if they log the errors
3. exit-early on simple cases pattern
4. hoping we can just return the godiscver error straight up, but maybe it's not very readable?